state/core/src/main/java/com/expediagroup/streamplatform/streamregistry/state/StateKey.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2018-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.state;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+
+@EqualsAndHashCode
+@ToString
+@AllArgsConstructor
+class StateKey {
```

Comment:
I had assumed the `deleted` flag would have been added to `Entity`, not creating a new wrapper key. I can't actually think of any reason why this is bad though. It does open us up to the potential bug of having both the deleted and undeleted key present at the same time though.

Reply:
I was trying to cover the case where while an Agent is offline, a delete is fired followed by another create using the same key. 

I didn't want an offline agent to miss that delete, but the more I'm playing with this implementation in an actual agent the more I'm realising that:
1. it's difficult for the agent to know what to do in that situation
2. the juice isn't worth the squeeze for handling what is probably going to be an extreme corner case.